python/docs/dlis/investigation.rst

```diff
@@ -0,0 +1,171 @@
+Investigation hints
+===================
+
+This section might be helpful to dlisio developers or very advanced users who
+want to find a cause of a specific dlisio failure.
+
+.. note::  Unfortunately, dlisio has little debugging support. Thus failure
+           investigation at the moment is not easy.
+
+.. note::  You **might** be able to fix the file by making changes
+           just in a couple of places.
+
+.. contents:: :local:
+
+
+Prerequisites
+-------------
+
+You:
+
+ 1. Are in extreme need of reading as much data as possible from That One File
+    which dlisio fails to read, or need to investigate it
+ 2. Are very familiar with rp66 specification
+ 3. Have a good grasp on working with binary files
+ 4. Can build dlisio_ and lfp_ from source code and have enough knowledge to
+    debug it
+
+
+Overview
+--------
+
+When error happens on load, it usually means that file structure itself is
+spoiled. So you should be able to find error just by checking VR and LR
+headers.
+
+When the file itself loads, it means that data layout is fine, which
+*probably* means that *data* itself is fine and is written as intended. Errors
+on accessing objects indicate that intention didn't match RP66 specification.
+
+
+Visual inspection
+-----------------
+
+Visually inspect the file to decide if failure cause is obvious.
+
+Open the file in binary editor, scroll through it.
+
+* Make sure that it looks like every other dlis file that opens correctly.
+  Regardless of the extension, file contents might not be dlis.
```

Comment:
Make sure it is a dlis file. Often the extensions lie. Look for the SUL or the first VR header